server/src/main/java/org/elasticsearch/index/codec/FieldInfosWithUsages.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.codec;
+
+import org.apache.lucene.index.DocValuesType;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.index.FieldInfos;
+import org.apache.lucene.index.IndexOptions;
+
+public class FieldInfosWithUsages extends FieldInfos {
+    private final int totalUsages;
+
+    public FieldInfosWithUsages(FieldInfo[] infos) {
+        super(infos);
+        this.totalUsages = computeUsages(infos);
+    }
+
+    public static int computeUsages(FieldInfo[] infos) {
+        int usages = 0;
+        for (FieldInfo fi : infos) {
+            if (fi.getIndexOptions() != IndexOptions.NONE) {
+                usages++;
+            }
+            if (fi.hasNorms()) {
+                usages++;
+            }
+            if (fi.hasVectors()) {
```

Comment:
We can probably skip term vectors: their memory usage does not scale with the number of fields (like stored fields).

Reply:
Yes, I have pushed https://github.com/elastic/elasticsearch/pull/112760/commits/5cc4f3524ad624d713dcc21cb0093cc252845d72 to remove this.